auth/credentials/api.go

```diff
@@ -28,14 +31,20 @@ type AtlasTokenSourceOptions struct {
 }
 
 // NewTokenSourceWithOptions initializes an OAuthTokenSource with advanced credentials.AtlasTokenSourceOptions
-// Use this method to initialize custom OAuth Token Cache (filesystem).
+// Use this method to initialize custom OAuth Token Cache (filesystem) or custom User Agent header.
```

Comment:
As this method evolves that comment might no longer be relevant. I made it with notion to flag potential use case but we will handle it by hand written docs.
```suggestion
```

Reply:
Yeah makes sense, I can remove it 👍